packages/deprecation-crawler/src/lib/commands/history.command.ts

```diff
@@ -1,40 +1,31 @@
-import { setup } from '../processors/setup';
 import { CrawledRelease, Deprecation, GitTag } from '../models';
 import {
   getCurrentBranchOrTag,
   git,
+  readRepoConfig,
   run,
+  updateRepoConfig,
   writeRawDeprecations,
 } from '../utils';
 import { YargsCommandObject } from '../cli/model';
 import { checkout } from '../tasks/checkout';
 import { crawl } from '../processors/crawl';
 import { addVersion } from '../tasks/add-version';
 import { prompt } from 'enquirer';
-import { existsSync } from 'fs';
-import { join } from 'path';
-import { RAW_DEPRECATION_PATH } from '../constants';
-import { logError } from '../log';
 import { commitChanges } from '../tasks/commit-changes';
 import { getTagChoices } from '../tasks/ensure-git-tag';
+import { ensureConfigDefaults } from '../tasks/ensure-config-defaults';
 
 export const historyCommand: YargsCommandObject = {
   command: 'history',
   description: 'Crawls for deprecations in previous versions',
   module: {
     handler: async (argv) => {
       if (argv.verbose) console.info(`run history as a yargs command`);
-      const config = await setup();
-
-      const rawDeprecationsPath = join(
-        config.outputDirectory,
-        `${RAW_DEPRECATION_PATH}`
-      );
-      if (existsSync(rawDeprecationsPath)) {
-        logError(
-          'The history command can only be run on a clean repository without existing deprecations'
-        );
-        return;
+      let config = readRepoConfig();
```

Comment:
Don't use the `setup` task because it saves the config to disk.
Because of it the checkout of a branch might throw an error.

An alternative is to remove the config file after the setup.
If we want to ask the setup questions to the users, we could also copy paste the ensures here:

```
  const config = {
    ...repoConfig,
    ...(await ensureDeprecationUrlConfig(repoConfig)
      .then(ensureDeprecationCommentConfig)
      .then(ensureOutputDirectoryConfig)
      .then(ensureIncludeGlobConfig)
      .then(ensureExcludeGlobConfig)
      // defaults should be last as it takes user settings
      .then(ensureConfigDefaults)),
  };
```

Reply:
I would go with the ensure questions and see if we can take the defaults?
e.g. ensureDeprecationUrlConfig will suggest the URL form package.json. 
Instead of Asking with this as suggestion we could just take it if present?

Maybe this works with and `--interactive` flag? This could also be used to test the ensure questions that can suggest values.

Reply:
See #117

Reply:
Yep, I like it.
I'll wait on that PR and make the changes in this branch afterwards.
Thanks for the review!

Reply:
@BioPhoton it's updated 😃